### PR TITLE
[OFE-318] - Updated the salary tracker to include a filter for depart…

### PR DESCRIPTION
**Pull request by KevJ2021**
…ments

### CODE REVIEW

File: _includes/salary-tracker.html [hunk 1]
```diff
@@ -161,16 +180,55 @@ <h5 class="mt-5 salary-tracker-info-label">Salary Brackets</h5>
...
+        var countryArr = dataArr.filter(obj => {
+            return obj.country === selectedCountry;
+        });
+        if (updateOption === 0) {
+            removeSelectOptions(selectRoleEle);
+            selectedPosition = null;
+        }
+
+        if (countryArr[0].positions[0] !== undefined) {
+            countryArr[0].positions.forEach(position => {
```
Comment (richmungla) [inline _includes/salary-tracker.html]: Don't forget to check if countryArr and positions is greater than 0 :) possible array out of bounds ?

File: _includes/salary-tracker.html [hunk 2]
```diff
@@ -179,43 +237,49 @@ <h5 class="mt-5 salary-tracker-info-label">Salary Brackets</h5>
...
         var countryArr = dataArr.filter(obj => {
             return obj.country === selectedCountry
         });
 
+        // Load values into Positions and Department Dropdowns
+        updateSelectOptions(updateOption);
+
+        // Proccess only if the country has positions
         if (countryArr[0].positions !== undefined) {
-            var benefitsEle = document.getElementById('salary-tracker-benefits');
```
Comment (richmungla) [inline _includes/salary-tracker.html]: Here too?

**Review by richmungla — APPROVED** [review]
Looks good to me 😄 👍🏾

**Comment by KevJ2021** [comment]
Thank you for the feedback Rich!

We shouldn't have an issue, as JS doesn't really have an index out of bounds exception (It returns undefined) - And we wouldn't be able to check the length either, as it will technically still have one element (Undefined) if the index is out of bounds.

But I do think that I need to add a check in for dataArr.countries length in main before executing everything else, thanks again!

**Pull request merged**